src/tab/Tab.tsx

```diff
@@ -49,6 +49,16 @@ function Tab({
   const [activeTabIndex, setActiveTabIndex] = useState(initialActiveTabIndex);
   const tabClassName = classNames("tab", customClassName);
 
+  let activeTabIndexToUse = activeTabIndexFromProps || activeTabIndex;
+  
+  /**
+   * In case activeTabIndex is out of bounds, we need to reset it to 0.
+   * This can happen if items are changes after the mount
+   */
+  if (activeTabIndexToUse > items.length) {
+    activeTabIndexToUse = 0;
+  }
```

Comment:
This should be reflected to the state and also we should call `onTabChange` if the Tab is being used as a controlled component